tests/integration/targets/aci_management_consumed_contract/tasks/main.yml

```diff
@@ -0,0 +1,197 @@
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: debug
+
+# CLEAN ENVIRONMENT
+- name: Remove Management Network Instance Profile
+  cisco.aci.aci_management_network_instance_profile:
+    <<: *aci_info
+    name: ans_network_inst_profile
+    state: absent
+
+- name: Remove OOB Contracts
+  cisco.aci.aci_oob_contract:
+    <<: *aci_info
+    contract: "{{ item }}"
+    state: absent
+  loop:
+    - ans_oob_ctr1
+    - ans_oob_ctr2
+
+# ADD CONTRACT BINDING
+- name: Create Management Network Instance Profile
+  cisco.aci.aci_management_network_instance_profile:
+    <<: *aci_info
+    name: ans_network_inst_profile
+    qos_class: unspecified
+    description: Ansible test profile
+    subnets:
+      - 10.20.30.0/24
+      - 192.168.10.0/24
+    state: present
+
+- name: Create OOB Contracts
+  cisco.aci.aci_oob_contract:
+    <<: *aci_info
+    contract: "{{ item }}"
+    state: present
+  loop:
+    - ans_oob_ctr1
+    - ans_oob_ctr2
+
+- name: Bind OOB Contracts To Profile - Check Mode
+  cisco.aci.aci_management_consumed_contract:
+    <<: *aci_info
+    profile: ans_network_inst_profile
+    consumed_contract: ans_oob_ctr1
+    state: present
+  check_mode: true
+  register: cm_add_contract_binding
+
+- name: Bind OOB Contracts To Profile - Normal Mode
+  cisco.aci.aci_management_consumed_contract:
+    <<: *aci_info
+    profile: ans_network_inst_profile
+    consumed_contract: ans_oob_ctr1
+    state: present
+  register: nm_add_contract_binding
+
+- name: Bind OOB Contracts To Profile Again - Test Idempotence
+  cisco.aci.aci_management_consumed_contract:
+    <<: *aci_info
+    profile: ans_network_inst_profile
+    consumed_contract: ans_oob_ctr1
+    state: present
+  register: nm_add_contract_binding_again
+
+- name: Verify Object Creation
+  ansible.builtin.assert:
+    that:
+      - cm_add_contract_binding is changed
+      - cm_add_contract_binding.previous == []
+      - cm_add_contract_binding.current == []
+      - cm_add_contract_binding.proposed.mgmtRsOoBCons.attributes.tnVzOOBBrCPName == "ans_oob_ctr1"
+      - nm_add_contract_binding is changed
+      - nm_add_contract_binding.previous == []
+      - nm_add_contract_binding.current.0.mgmtRsOoBCons.attributes.tnVzOOBBrCPName == "ans_oob_ctr1"
+      - nm_add_contract_binding.proposed == cm_add_contract_binding.proposed
+      - nm_add_contract_binding_again is not changed
+      - nm_add_contract_binding_again.previous == nm_add_contract_binding.current
+      - nm_add_contract_binding_again.current == nm_add_contract_binding.current
+      - nm_add_contract_binding_again.proposed == nm_add_contract_binding.proposed
+
+# MODIFY CONTRACT BINDING
+- name: Update OOB Contract Binding
+  cisco.aci.aci_management_consumed_contract:
+    <<: *aci_info
+    profile: ans_network_inst_profile
+    consumed_contract: ans_oob_ctr1
+    dscp_priority: level1
+    state: present
+  register: update_contract_binding
```

Comment:
can you register with cm_ , nm_, and nm_..._again and assert accordingly. This also applies for delete